### PR TITLE
v0.9.0.1

### PR DESCRIPTION
**Pull request by jjr840430**
Integrated configuration module with `go-multiconfig`

### CODE REVIEW

File: common.go [hunk 1]
```diff
@@ -48,6 +48,167 @@ const (
 	KAOHI_DEFAULT_SYSLOG_PROTO       = "tcp"
 )
 
+const (
+	KAOHI_HCL_OPTIONS = `cmdline "log_directory" {
```
Comment (ronaldtse) [inline common.go]: @jjr840430 can you extract this out to a separate file, called config.mel

Comment (jjr840430) [reply common.go]: should `config.mel` be located in the same directory with executable?

Comment (jjr840430) [reply common.go]: @ronaldtse Fixed, I added `config.mel` in `./bin` directory.

Comment (ronaldtse) [reply common.go]: @jjr840430 not really -- `config.mel` should only be available in compilation, the user doesn't need it, so parse and store the contents of the file into the executable.

Comment (jjr840430) [reply common.go]: when I included `config.mel` to be compiled with `.go` files, the compiler said:

```
named files must be .go files
```

Comment (ronaldtse) [reply common.go]: There must be some way you can pre-parse the mel file into Go, and then compile Kaohi?

**Comment by ronaldtse** [comment]
@jjr840430 just want to add that there are two parts to using go-multiconfig:

1. During compiling, the precompiling-part of go-multiconfig should parse and add the content of `config.mel` into Kaohi as static Go code. (i.e. go-multiconfig should "convert" `config.mel` into `config.mel.go`, just like `autoconf` in C)

2. After compiling, the run-time part of the go-multiconfig library will provide the run-time functionality like help, env var parsing, etc. We will not distribute the executable with `config.mel`.

**Comment by ronaldtse** [comment]
@jjr840430 I'm merging this first, but please see the above comment.

**Pull request merged**